lib/matplotlib/contour.py

```diff
@@ -1371,7 +1371,7 @@ def find_nearest_contour(self, x, y, indices=None, pixel=True):
         # Nonetheless, improvements could probably be made.
 
         if indices is None:
-            indices = range(len(self.levels))
+            indices = range(len(self.layers))
```

Comment:
I have a slight preference to change this to

```suggestion
            indices = range(len(self.collections))
```

as that is the data structure that we need to match?

Reply:
I think this change makes sense. I outlined some issues below that I'd like you to look at.